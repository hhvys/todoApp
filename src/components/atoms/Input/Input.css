input {
    background-color: inherit;
    border: none;
    padding-left: 20px;
    font-size: 16px;
    flex: 12;
    border-radius: 5px;
    color: inherit;
}

input::placeholder {
    color: #FFFFFF;
    opacity: 0.6;
}