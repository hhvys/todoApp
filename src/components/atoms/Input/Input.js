import React from 'react';
import styles from './Input.mod.scss';

const handleChange = (e, onInputChange) => {
	onInputChange(e.target.value);
};

const Input = ({className, onInputChange, inputValue, ...props}) => {
	return (<input
		className={`${className ? className : ''} ${styles.input} full-height d-flex align-items-center justify-content-center`}
		onChange={(e) => handleChange(e, onInputChange)}
		value={inputValue}
		{...props}
	/>);
};

export default Input;
