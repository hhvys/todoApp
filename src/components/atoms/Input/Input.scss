.input {
    background-color: inherit;
    border: none;
    font-size: 16px;
    flex: 12;
    border-radius: 5px;
    color: inherit;
}

.input::placeholder {
    color: #FFFFFF;
    opacity: 0.6;
}
