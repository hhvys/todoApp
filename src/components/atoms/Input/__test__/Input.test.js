import React from 'react';
import Input from '../Input';

describe('component', () => {
	describe('Input', () => {
		it('should render', () => {
			expect(toJSON(shallow(<Input/>))).toMatchSnapshot();
		});
		it('should render className', () => {
			const props = {
				className: 'className'
			};
			expect(toJSON(shallow(<Input {...props}/>))).toMatchSnapshot();
		});
		it('should pass inputValue', () => {
			const props = {
				inputValue: 'text'
			};
			expect(toJSON(shallow(<Input {...props}/>))).toMatchSnapshot();
		});
	});
});
