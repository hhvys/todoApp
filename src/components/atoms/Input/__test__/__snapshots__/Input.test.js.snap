// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component Input should pass inputValue 1`] = `
<input
  className=" input full-height d-flex align-items-center justify-content-center"
  onChange={[Function]}
  value="text"
/>
`;

exports[`component Input should render 1`] = `
<input
  className=" input full-height d-flex align-items-center justify-content-center"
  onChange={[Function]}
/>
`;

exports[`component Input should render className 1`] = `
<input
  className="className input full-height d-flex align-items-center justify-content-center"
  onChange={[Function]}
/>
`;
